### PR TITLE
Fine-tuning on AMY for diurnal cycles + export

### PR DESCRIPTION
**Pull request by vicford**
This PR works explicitly for in-notebook features for the Average Meteorological year notebook: https://github.com/cal-adapt/cae-examples/blob/typical_meteorological_year/explore_amy.ipynb

Effectively: the two critical AMY functions **tmy_calc** and **remove_repeats** were pulled out of the class so that they could be calleable in-notebook for analyzing/plotting a diurnal cycle, an annual cycle, and export functionality of the AMY dataframe itself. Some of the dataframe transforms were therefore included and cleaned up in the larger script. 

**Note**: Exporting the AMY dataframe _currently_ does not conform to our existing export functionality, so there is a separate in line call for that. Exporting the relevant catalog data works as desired. 

To test this: please run through the explore_amy notebook over in cae-examples branch `typical_meteorological_year`

### CODE REVIEW

**Review by nicolejkeeney — APPROVED** [review]
This looks great! I tested the notebook and had no issues. Plots look really good.

File: climakitae/tmy.py
Comment (elehmer) [inline climakitae/tmy.py:137]: This can be rewritten as:

```
cols = df_amy.columns.tolist()
cols = cols[7:] + cols[:7]
df_amy = df_amy[cols]
```

File: climakitae/tmy.py
Comment (elehmer) [inline climakitae/tmy.py:138]: This can be rewritten as:
```
n_col_lst = []
for ampm in ['am', 'pm']:
    hr_lst = []
    for hr in range(1, 13, 1):
        hr_lst.append(str(hr)+ampm)
    hr_lst = hr_lst[-1:] + hr_lst[:-1]
    n_col_lst = n_col_lst + hr_lst
df_amy.columns = n_col_lst
```

**Review by elehmer — CHANGES_REQUESTED** [review]
I suggested a couple of programmatic alternatives to hard coding the time shifts. I wonder if there is a TimeZone object we could use to do this shift instead without so much code. But for now this works.

File: climakitae/tmy.py
Comment (vicford) [reply climakitae/tmy.py]: Done!

File: climakitae/tmy.py
Comment (vicford) [reply climakitae/tmy.py]: Done!

**Review by elehmer — APPROVED** [review]
LGTM!! When I mocked up the changes I assumed hourly_list had the shape of 24x365[6]. If that is correct then all should be good!

**Comment by elehmer** [comment]
We should probably test this branch to see if it produces the same results before merging...

**Comment by vicford** [comment]
> test this branch to see if it produces the same results before merging

Checked on the hub -- no issues!

**Comment by elehmer** [comment]
> > test this branch to see if it produces the same results before merging
> 
> Checked on the hub -- no issues!

Hmm, I got this error at Step 2:

```
ImportError                               Traceback (most recent call last)
/tmp/ipykernel_728/275966440.py in <module>
----> 1 from climakitae.tmy import tmy_calc
      2 import pandas as pd
      3 import numpy as np
      4 import datetime

ImportError: cannot import name 'tmy_calc' from 'climakitae.tmy' (/home/jovyan/src/climakitae/climakitae/tmy.py)
```

**Comment by vicford** [comment]
> running the right code base

This is what works for me: pip install -e git+https://github.com/cal-adapt/climakitae.git@amy_finetune#egg=climakitae

**Comment by elehmer** [comment]
> > running the right code base
> 
> This is what works for me: pip install -e git+https://github.com/cal-adapt/climakitae.git@amy_finetune#egg=climakitae

Ok, looks good. Go for merge!!

**Pull request merged**
